layer/main/asset.py

```diff
@@ -0,0 +1,281 @@
+import logging
+from typing import TYPE_CHECKING, Any, Callable, Optional
+
+from layer.cache.utils import is_cached
+from layer.clients.layer import LayerClient
+from layer.config import ConfigManager
+from layer.config.config import Config
+from layer.context import Context
+from layer.contracts.assets import AssetPath, AssetType
+from layer.contracts.datasets import Dataset
+from layer.contracts.models import Model
+from layer.contracts.project_full_name import ProjectFullName
+from layer.contracts.tracker import ResourceTransferState
+from layer.exceptions.exceptions import ProjectInitializationException
+from layer.global_context import (
+    current_account_name,
+    get_active_context,
+    reset_active_context,
+    set_active_context,
+)
+from layer.projects.utils import get_current_project_full_name
+from layer.tracker.utils import get_progress_tracker
+from layer.utils.async_utils import asyncio_run_in_thread
+
+from .utils import sdk_function
+
+
+if TYPE_CHECKING:
+    import pandas
+
+logger = logging.getLogger(__name__)
+
+
+@sdk_function
```

Comment:
Would this also trigger if inside the executors? Currently users have no control over what version of the SDK is built-in the image so a warning will be confusing if they already have the latest version.

Reply:
Wouldn't it be simpler if we issue that warning only when people do `import layer`?

Reply:
This is not changing functionality so much as moving it to a decorator. We were already checking things inside each function.

Reply:
> This is not changing functionality so much as moving it to a decorator. We were already checking things inside each function.

The bit that changed though is that before it was checking only on `layer.login` and `layer.init` which are not called inside the executors right? Now we do it for all sdk functions.